lib/journald/src/lib.rs

```diff
@@ -107,6 +111,21 @@ impl Journal {
         })
         .collect::<IOResult<Record>>()
     }
+
+    pub fn cursor(&self) -> IOResult<String> {
+        let mut cursor: *mut c_char = null_mut();
+        sd_result(self.lib.sd_journal_get_cursor(self.journal, &mut cursor))?;
+        Ok(into_string(cursor).unwrap())
+    }
+
+    pub fn seek_cursor(&self, cursor: &str) -> IOResult<()> {
```

Comment:
seems odd that we are passing around a str for a cursor is there any more type level stuff we could add to this?

Reply:
The cursor from libsystemd is a string. Do you mean to wrap it in a dummy struct?